test/integration/deprecated_specs.cc

```diff
@@ -38,3 +42,89 @@ TEST(DeprecatedSpecs, Spec1_2)
+TEST(DeprecatedElements, CanEmitErrors)
+{
+  sdf::SDFPtr sdf(new sdf::SDF());
+  sdf::init(sdf);
+  auto elem = std::make_shared<sdf::Element>();
+  elem->SetRequired("-1");
+  elem->SetName("testElem");
+  // Change the root of sdf so we can test "testElem" in isolation
+  sdf->Root(elem);
+  auto config = sdf::ParserConfig::GlobalConfig();
+  config.SetWarningsPolicy(sdf::EnforcementPolicy::ERR);
+  sdf::Errors errors;
+  EXPECT_TRUE(sdf::readString(
+      "<sdf version='1.8'><testElem/></sdf>", config, sdf, errors));
+  ASSERT_FALSE(errors.empty());
+  EXPECT_EQ(sdf::ErrorCode::ELEMENT_DEPRECATED, errors[0].Code());
+}
+
+////////////////////////////////////////////////////
+TEST(DeprecatedElements, CanEmitWarningWithErrorEnforcmentPolicy)
+{
+  // Capture sdfwarn output, which is treamed to std::cerr
+  std::stringstream buffer;
+  auto old = std::cerr.rdbuf(buffer.rdbuf());
+
+#ifdef _WIN32
+  sdf::Console::Instance()->SetQuiet(false);
+#endif
+
+  auto contains = [](const std::string &_a, const std::string &_b)
+  {
+    return _a.find(_b) != std::string::npos;
+  };
+
+  sdf::SDFPtr sdf(new sdf::SDF());
+  sdf::init(sdf);
+  auto elem = std::make_shared<sdf::Element>();
+  elem->SetRequired("-1");
+  elem->SetName("testElem");
+  // Change the root of sdf so we can test "testElem" in isolation
+  sdf->Root(elem);
+  {
+    auto config = sdf::ParserConfig::GlobalConfig();
+    config.SetWarningsPolicy(sdf::EnforcementPolicy::ERR);
+    config.SetDeprecatedElementsPolicy(sdf::EnforcementPolicy::WARN);
+    sdf::Errors errors;
+    EXPECT_TRUE(sdf::readString(
+          "<sdf version='1.8'><testElem/></sdf>", config, sdf, errors));
+    EXPECT_TRUE(errors.empty());
+    EXPECT_PRED2(contains, buffer.str(), "SDF Element[testElem] is deprecated");
+  }
+  // Flip the order of calling SetDeprecatedElementsPolicy and
+  // SetWarningsPolicy.
+  {
+    // clear the contents of the buffer
+    buffer.str("");
+    auto config = sdf::ParserConfig::GlobalConfig();
+    config.SetDeprecatedElementsPolicy(sdf::EnforcementPolicy::WARN);
+    config.SetWarningsPolicy(sdf::EnforcementPolicy::ERR);
+    sdf::Errors errors;
+    EXPECT_TRUE(sdf::readString(
+          "<sdf version='1.8'><testElem/></sdf>", config, sdf, errors));
+    EXPECT_TRUE(errors.empty());
+    EXPECT_PRED2(contains, buffer.str(), "SDF Element[testElem] is deprecated");
+  }
+  // Test ResetDeprecatedElementsPolicy
+  {
+    auto config = sdf::ParserConfig::GlobalConfig();
+    config.SetDeprecatedElementsPolicy(sdf::EnforcementPolicy::WARN);
+    config.SetWarningsPolicy(sdf::EnforcementPolicy::ERR);
+    config.ResetDeprecatedElementsPolicy();
+    sdf::Errors errors;
+    EXPECT_TRUE(sdf::readString(
+          "<sdf version='1.8'><testElem/></sdf>", config, sdf, errors));
+    ASSERT_FALSE(errors.empty());
+    EXPECT_EQ(sdf::ErrorCode::ELEMENT_DEPRECATED, errors[0].Code());
+  }
+
+  // Revert cerr rdbug so as to not interfere with other tests
+  std::cerr.rdbuf(old);
```

Comment:
nit This may not be reachable in test failure which interrupts execution (usage of `ASSERT_*`).

Consider a C++ RAII pattern. Example from Drake:
https://github.com/RobotLocomotion/drake/blob/v0.29.0/common/scope_exit.h

Reply:
Added `ScopeExit` in https://github.com/osrf/sdformat/pull/543/commits/eebf49b92f85392d042044151aeb7206c18cf079.

Reply:
Sweet!